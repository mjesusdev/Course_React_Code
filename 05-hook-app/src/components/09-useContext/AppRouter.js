import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from 'react-router-dom';

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                
                <NavBar />

                <div className="container">
                    <Switch>
                        <Route exact path="/" component={ HomeScreen } />

                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />
                        
                        <Redirect to="/" />

                    </Switch>
                </div>
            </div>
        </Router>
    )
}
