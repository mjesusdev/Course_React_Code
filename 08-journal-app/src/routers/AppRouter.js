import React from 'react';
import {
    BrowserRouter as Router,
    Redirect,
    Route,
    Switch
} from 'react-router-dom';

import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch> 
                    <Route  
                        path="/auth" 
                        component={ AuthRouter } 
                    />
                    
                    <Route 
                        exact
                        path="/"
                        component={ JournalScreen }
                    />

                    <Redirect to="/auth/login" />
                    
                </Switch>
            </div>
        </Router>
    )
}
