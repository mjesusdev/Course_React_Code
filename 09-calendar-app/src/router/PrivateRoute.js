import React from 'react';
import PropTypes from 'prop-types';

import { Route, Redirect } from 'react-router';

export const PrivateRoute = ({
    isAutenticated,
    component: Component,
    ...rest
}) => {

    return (
        <Route { ...rest } 
            component={ (props) => (
                ( isAutenticated )
                    ? ( <Component { ...props } /> )
                    : ( <Redirect to="/login" />)
            )}
        />
    )
}

PrivateRoute.propTypes = {
    isAutenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}