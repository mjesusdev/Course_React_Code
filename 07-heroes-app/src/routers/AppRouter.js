import React, { useContext } from 'react'
import { Routes } from 'react-router-dom';

import { AuthContext } from '../auth/AuthContext';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {

    const { user } = useContext(AuthContext);

    return (
        <div>
            <Routes> 
                <PublicRoute
                    path="/login"
                    element={ <LoginScreen /> } 
                    isAutenticated={ user.logged }
                />
                
                <PrivateRoute 
                    path="/*"
                    element={ <DashboardRoutes /> }
                    isAutenticated={ user.logged }
                />
            </Routes>
        </div>
    )
}
