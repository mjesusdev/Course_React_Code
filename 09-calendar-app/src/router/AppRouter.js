import React, { useEffect } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Redirect
} from "react-router-dom";

import { useDispatch, useSelector } from 'react-redux';

import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { startChecking } from '../actions/auth';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const dispatch = useDispatch();
    const { checking, uid } = useSelector(state => state.auth);

    useEffect(() => {

        dispatch(startChecking() );

    }, [dispatch])

    if ( checking ) {
        return (<h5>Wait...</h5>);
    }

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute 
                        exact 
                        path="/login"
                        component={ LoginScreen } 
                        isAutenticated={ !!uid }
                    />
                    
                    <PrivateRoute 
                        exact 
                        path="/" 
                        component={ CalendarScreen } 
                        isAutenticated={ !!uid }
                    />

                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    )
}