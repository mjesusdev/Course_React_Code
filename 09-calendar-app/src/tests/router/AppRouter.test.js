import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { AppRouter } from '../../router/AppRouter';

// Configure Thunk and Store
const middlewars = [thunk];
const mockStore = configureStore(middlewars);

/* store.dispatch = jest.fn(); */

describe('Test in <AppRouter />', () => {
    
    test('should shows the first step - Waiting', () => {
        const initState = {
            auth: {
                checking: 'true'
            }
        };
        const store = mockStore(initState);
        
        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h5').exists() ).toBe(true);
    })

    test('should shows the public route âœ…', () => {
        const initState = {
            auth: {
                checking: false,
                uid: null
            }
        };
        const store = mockStore(initState);

        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.login-container').exists() ).toBe(true);
    })

    test('should shows the CalendarScreen ðŸ“—', () => {
        const initState = {
            calendar: {
                events: []
            },
            ui: {
                modalOpen: false
            },
            auth: {
                checking: false,
                uid: '123',
                name: 'Test'
            }
        };
        const store = mockStore(initState);

        const wrapper = mount(
            <Provider store={ store }>
                <AppRouter />
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.calendar-screen').exists() ).toBe(true);
    })
    
})
