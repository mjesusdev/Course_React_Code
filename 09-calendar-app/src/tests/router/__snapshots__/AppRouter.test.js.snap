// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter /> should shows the CalendarScreen ðŸ“— 1`] = `
<div>
  <div
    className="calendar-screen"
  >
    <div
      className="navbar navbar-dark bg-dark mb-4"
    >
      <span
        className="navbar-brand"
      >
        Test
      </span>
      <button
        className="btn btn-outline-danger"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        <span>
           Exit
        </span>
      </button>
    </div>
    <div
      className="rbc-calendar"
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Today
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          June 2021
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Month
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Week
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Day
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        aria-label="Month View"
        className="rbc-month-view"
        role="table"
      >
        <div
          className="rbc-row rbc-month-header"
          role="row"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sun
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Mon
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Tue
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Wed
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Thu
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Fri
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sat
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  31
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  05
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  07
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  09
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  10
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  11
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  12
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-today"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  14
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  16
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  17
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  18
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  19
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  21
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  23
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  24
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  25
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  26
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="btn btn-primary fab"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </button>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
  </div>
</div>
`;

exports[`Test in <AppRouter /> should shows the first step - Waiting 1`] = `
<h5>
  Wait...
</h5>
`;

exports[`Test in <AppRouter /> should shows the public route âœ… 1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Login
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lEmail"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value="hola@jesus.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="123456789"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rName"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value="Nando"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rEmail"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value="nando@gmail.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword1"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="12345678"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword2"
              onChange={[Function]}
              placeholder="Repeat the password"
              type="password"
              value="12345678"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
